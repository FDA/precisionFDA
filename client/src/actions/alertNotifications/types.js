export const ALERT_SHOW_ABOVE_ALL = 'ALERT_SHOW_ABOVE_ALL'
export const ALERT_HIDE_MESSAGE = 'ALERT_HIDE_MESSAGE'
