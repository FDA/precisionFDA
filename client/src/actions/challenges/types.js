export const CHALLENGES_FETCH_START   = 'CHALLENGES_FETCH_START'
export const CHALLENGES_FETCH_SUCCESS = 'CHALLENGES_FETCH_SUCCESS'
export const CHALLENGES_FETCH_FAILURE = 'CHALLENGES_FETCH_FAILURE'

export const CHALLENGES_YEAR_LIST_FETCH_START = 'CHALLENGES_YEAR_LIST_FETCH_START'
export const CHALLENGES_YEAR_LIST_FETCH_SUCCESS = 'CHALLENGES_YEAR_LIST_FETCH_SUCCESS'
export const CHALLENGES_YEAR_LIST_FETCH_FAILURE = 'CHALLENGES_YEAR_LIST_FETCH_FAILURE'

export const CHALLENGE_FETCH_START    = 'CHALLENGE_FETCH_START'
export const CHALLENGE_FETCH_SUCCESS  = 'CHALLENGE_FETCH_SUCCESS'
export const CHALLENGE_FETCH_FAILURE  = 'CHALLENGE_FETCH_FAILURE'

export const CHALLENGES_SET_PAGE = 'CHALLENGES_SET_PAGE'
export const CHALLENGES_SET_YEAR = 'CHALLENGES_SET_YEAR'
export const CHALLENGES_SET_TIME_STATUS = 'CHALLENGES_SET_TIME_STATUS'
export const CHALLENGES_LIST_RESET_FILTERS = 'CHALLENGES_LIST_RESET_FILTERS'

export const PROPOSE_CHALLENGE_FETCH_START = 'PROPOSE_CHALLENGE_FETCH_START'
export const PROPOSE_CHALLENGE_FETCH_SUCCESS = 'PROPOSE_CHALLENGE_FETCH_SUCCESS'
export const PROPOSE_CHALLENGE_FETCH_FAILURE = 'PROPOSE_CHALLENGE_FETCH_FAILURE'
export const PROPOSE_CHALLENGE_FORM_RESET = 'PROPOSE_CHALLENGE_FORM_RESET'
