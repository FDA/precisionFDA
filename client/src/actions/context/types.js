export const CONTEXT_FETCH_START = 'CONTEXT_FETCH_START'
export const CONTEXT_FETCH_SUCCESS = 'CONTEXT_FETCH_SUCCESS'
export const CONTEXT_FETCH_FAILURE = 'CONTEXT_FETCH_FAILURE'
