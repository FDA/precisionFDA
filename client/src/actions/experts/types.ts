export const EXPERTS_LIST_FETCH_START   = 'EXPERTS_LIST_FETCH_START'
export const EXPERTS_LIST_FETCH_SUCCESS = 'EXPERTS_LIST_FETCH_SUCCESS'
export const EXPERTS_LIST_FETCH_FAILURE = 'EXPERTS_LIST_FETCH_FAILURE'
export const EXPERTS_LIST_SET_PAGE = 'EXPERTS_LIST_SET_PAGE'
export const EXPERTS_LIST_SET_YEAR = 'EXPERTS_LIST_SET_YEAR'
export const EXPERTS_LIST_RESET_FILTERS = 'EXPERTS_LIST_RESET_FILTERS'

export const EXPERTS_YEAR_LIST_FETCH_START = 'EXPERTS_YEAR_LIST_FETCH_START'
export const EXPERTS_YEAR_LIST_FETCH_SUCCESS = 'EXPERTS_YEAR_LIST_FETCH_SUCCESS'
export const EXPERTS_YEAR_LIST_FETCH_FAILURE = 'EXPERTS_YEAR_LIST_FETCH_FAILURE'

export const EXPERTS_SHOW_MODAL = 'EXPERTS_SHOW_MODAL'
export const EXPERTS_HIDE_MODAL = 'EXPERTS_HIDE_MODAL'
