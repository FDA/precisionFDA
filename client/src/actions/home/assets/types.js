export const HOME_ASSETS_FETCH_START = 'HOME_ASSETS_FETCH_START'
export const HOME_ASSETS_FETCH_SUCCESS = 'HOME_ASSETS_FETCH_SUCCESS'
export const HOME_ASSETS_FETCH_FAILURE = 'HOME_ASSETS_FETCH_FAILURE'
export const HOME_ASSETS_TOGGLE_ALL_CHECKBOXES = 'HOME_ASSETS_TOGGLE_ALL_CHECKBOXES'
export const HOME_ASSETS_TOGGLE_CHECKBOX = 'HOME_ASSETS_TOGGLE_CHECKBOX'
export const HOME_ASSETS_SET_FILTER_VALUE = 'HOME_ASSETS_SET_FILTER_VALUE'
export const HOME_ASSETS_RESET_FILTERS = 'HOME_ASSETS_RESET_FILTERS'
export const HOME_ASSETS_FETCH_ASSET_DETAILS_START = 'HOME_ASSETS_FETCH_ASSET_DETAILS_START'
export const HOME_ASSETS_FETCH_ASSET_DETAILS_SUCCESS = 'HOME_ASSETS_FETCH_ASSET_DETAILS_SUCCESS'
export const HOME_ASSETS_FETCH_ASSET_DETAILS_FAILURE = 'HOME_ASSETS_FETCH_ASSET_DETAILS_FAILURE'
export const HOME_ASSETS_MAKE_FEATURED_SUCCESS = 'HOME_ASSETS_MAKE_FEATURED_SUCCESS'

export const HOME_ASSETS_SHOW_MODAL = 'HOME_ASSETS_SHOW_MODAL'
export const HOME_ASSETS_HIDE_MODAL = 'HOME_ASSETS_HIDE_MODAL'

export const HOME_ASSETS_MODAL_ACTION_START = 'HOME_ASSETS_MODAL_ACTION_START'
export const HOME_ASSETS_MODAL_ACTION_SUCCESS = 'HOME_ASSETS_MODAL_ACTION_SUCCESS'
export const HOME_ASSETS_MODAL_ACTION_FAILURE = 'HOME_ASSETS_MODAL_ACTION_FAILURE'
