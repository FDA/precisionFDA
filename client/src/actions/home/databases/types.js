export const HOME_DATABASES_FETCH_START = 'HOME_DATABASES_FETCH_START'
export const HOME_DATABASES_FETCH_SUCCESS = 'HOME_DATABASES_FETCH_SUCCESS'
export const HOME_DATABASES_FETCH_FAILURE = 'HOME_DATABASES_FETCH_FAILURE'

export const HOME_DATABASES_FETCH_DETAILS_START = 'HOME_DATABASES_FETCH_DETAILS_START'
export const HOME_DATABASES_FETCH_DETAILS_SUCCESS = 'HOME_DATABASES_FETCH_DETAILS_SUCCESS'
export const HOME_DATABASES_FETCH_DETAILS_FAILURE = 'HOME_DATABASES_FETCH_DETAILS_FAILURE'

export const HOME_DATABASES_TOGGLE_ALL_CHECKBOXES = 'HOME_DATABASES_TOGGLE_ALL_CHECKBOXES'
export const HOME_DATABASES_TOGGLE_CHECKBOX = 'HOME_DATABASES_TOGGLE_CHECKBOX'

export const HOME_DATABASES_EDIT_TAGS_START = 'HOME_DATABASES_EDIT_TAGS_START'
export const HOME_DATABASES_EDIT_TAGS_SUCCESS = 'HOME_DATABASES_EDIT_TAGS_SUCCESS'
export const HOME_DATABASES_EDIT_TAGS_FAILURE = 'HOME_DATABASES_EDIT_TAGS_FAILURE'

export const HOME_DATABASE_CREATE_START = 'HOME_DATABASE_CREATE_START'
export const HOME_DATABASE_CREATE_SUCCESS = 'HOME_DATABASE_CREATE_SUCCESS'
export const HOME_DATABASE_CREATE_FAILURE = 'HOME_DATABASE_CREATE_FAILURE'

export const HOME_DATABASES_SET_FILTER_VALUE = 'HOME_DATABASES_SET_FILTER_VALUE'
export const HOME_DATABASES_RESET_FILTERS = 'HOME_DATABASES_RESET_FILTERS'

export const HOME_DATABASES_SHOW_MODAL = 'HOME_DATABASES_SHOW_MODAL'
export const HOME_DATABASES_HIDE_MODAL = 'HOME_DATABASES_HIDE_MODAL'
export const HOME_DATABASES_RESET_MODALS = 'HOME_DATABASES_RESET_MODALS'

export const HOME_DATABASE_MODAL_ACTION_START = 'HOME_DATABASE_MODAL_ACTION_START'
export const HOME_DATABASE_MODAL_ACTION_SUCCESS = 'HOME_DATABASE_MODAL_ACTION_SUCCESS'
export const HOME_DATABASE_MODAL_ACTION_FAILURE = 'HOME_DATABASE_MODAL_ACTION_FAILURE'

export const HOME_COPY_DATABASE_TO_SPACE_START = 'HOME_COPY_DATABASE_TO_SPACE_START'
export const HOME_COPY_DATABASE_TO_SPACE_SUCCESS = 'HOME_COPY_DATABASE_TO_SPACE_SUCCESS'
export const HOME_COPY_DATABASE_TO_SPACE_FAILURE = 'HOME_COPY_DATABASE_TO_SPACE_FAILURE'

export const HOME_DATABASE_EDIT_INFO_START = 'HOME_DATABASE_EDIT_INFO_START'
export const HOME_DATABASE_EDIT_INFO_SUCCESS = 'HOME_DATABASE_EDIT_INFO_SUCCESS'
export const HOME_DATABASE_EDIT_INFO_FAILURE = 'HOME_DATABASE_EDIT_INFO_FAILURE'

export const HOME_DATABASES_RUN_ACTION_START = 'HOME_DATABASES_RUN_ACTION_START'
export const HOME_DATABASES_RUN_ACTION_SUCCESS = 'HOME_DATABASES_RUN_ACTION_SUCCESS'
export const HOME_DATABASES_RUN_ACTION_FAILURE = 'HOME_DATABASES_RUN_ACTION_FAILURE'
