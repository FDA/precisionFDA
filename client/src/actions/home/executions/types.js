export const HOME_EXECUTIONS_FETCH_START = 'HOME_EXECUTIONS_FETCH_START'
export const HOME_EXECUTIONS_FETCH_SUCCESS = 'HOME_EXECUTIONS_FETCH_SUCCESS'
export const HOME_EXECUTIONS_FETCH_FAILURE = 'HOME_EXECUTIONS_FETCH_FAILURE'
export const HOME_EXECUTIONS_TOGGLE_ALL_CHECKBOXES = 'HOME_EXECUTIONS_TOGGLE_ALL_CHECKBOXES'
export const HOME_EXECUTIONS_TOGGLE_CHECKBOX = 'HOME_EXECUTIONS_TOGGLE_CHECKBOX'
export const HOME_EXECUTIONS_EXPAND_EXECUTION = 'HOME_EXECUTIONS_EXPAND_EXECUTION'
export const HOME_EXECUTIONS_EXPAND_ALL_EXECUTION = 'HOME_EXECUTIONS_EXPAND_ALL_EXECUTION'
export const HOME_EXECUTIONS_FETCH_EXECUTION_DETAILS_START = 'HOME_EXECUTIONS_FETCH_EXECUTION_DETAILS_START'
export const HOME_EXECUTIONS_FETCH_EXECUTION_DETAILS_SUCCESS = 'HOME_EXECUTIONS_FETCH_EXECUTION_DETAILS_SUCCESS'
export const HOME_EXECUTIONS_FETCH_EXECUTION_DETAILS_FAILURE = 'HOME_EXECUTIONS_FETCH_EXECUTION_DETAILS_FAILURE'
export const HOME_EXECUTIONS_SET_FILTER_VALUE = 'HOME_EXECUTIONS_SET_FILTER_VALUE'
export const HOME_EXECUTIONS_RESET_FILTERS = 'HOME_EXECUTIONS_RESET_FILTERS'
export const HOME_EXECUTIONS_MAKE_FEATURED_SUCCESS = 'HOME_EXECUTIONS_MAKE_FEATURED_SUCCESS'

export const HOME_EXECUTIONS_SHOW_MODAL = 'HOME_EXECUTIONS_SHOW_MODAL'
export const HOME_EXECUTIONS_HIDE_MODAL = 'HOME_EXECUTIONS_HIDE_MODAL'

export const HOME_EXECUTION_MODAL_ACTION_START = 'HOME_EXECUTION_MODAL_ACTION_START'
export const HOME_EXECUTION_MODAL_ACTION_SUCCESS = 'HOME_EXECUTION_MODAL_ACTION_SUCCESS'
export const HOME_EXECUTION_MODAL_ACTION_FAILURE = 'HOME_EXECUTION_MODAL_ACTION_FAILURE'
