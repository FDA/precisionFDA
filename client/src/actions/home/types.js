export const HOME_APPS_FETCH_START = 'HOME_APPS_FETCH_START'
export const HOME_APPS_FETCH_SUCCESS = 'HOME_APPS_FETCH_SUCCESS'
export const HOME_APPS_FETCH_FAILURE = 'HOME_APPS_FETCH_FAILURE'
export const HOME_APPS_TOGGLE_ALL_CHECKBOXES = 'HOME_APPS_TOGGLE_ALL_CHECKBOXES'
export const HOME_APPS_TOGGLE_CHECKBOX = 'HOME_APPS_TOGGLE_CHECKBOX'
export const HOME_APPS_FETCH_APP_DETAILS_START = 'HOME_APPS_FETCH_APP_DETAILS_START'
export const HOME_APPS_FETCH_APP_DETAILS_SUCCESS = 'HOME_APPS_FETCH_APP_DETAILS_SUCCESS'
export const HOME_APPS_FETCH_APP_DETAILS_FAILURE = 'HOME_APPS_FETCH_APP_DETAILS_FAILURE'
export const HOME_APPS_DELETE_START = 'HOME_APPS_DELETE_START'
export const HOME_APPS_DELETE_SUCCESS = 'HOME_APPS_DELETE_SUCCESS'
export const HOME_APPS_DELETE_FAILURE = 'HOME_APPS_DELETE_FAILURE'
export const HOME_COPY_APP_TO_SPACE_START = 'HOME_COPY_APP_TO_SPACE_START'
export const HOME_COPY_APP_TO_SPACE_SUCCESS = 'HOME_COPY_APP_TO_SPACE_SUCCESS'
export const HOME_COPY_APP_TO_SPACE_FAILURE = 'HOME_COPY_APP_TO_SPACE_FAILURE'
export const HOME_ASSIGN_TO_CHALLENGE_START = 'HOME_ASSIGN_TO_CHALLENGE_START'
export const HOME_ASSIGN_TO_CHALLENGE_SUCCESS = 'HOME_ASSIGN_TO_CHALLENGE_SUCCESS'
export const HOME_ASSIGN_TO_CHALLENGE_FAILURE = 'HOME_ASSIGN_TO_CHALLENGE_FAILURE'
export const HOME_EDIT_WORKFLOW_TAGS_START = 'HOME_EDIT_WORKFLOW_TAGS_START'
export const HOME_EDIT_WORKFLOW_TAGS_SUCCESS = 'HOME_EDIT_WORKFLOW_TAGS_SUCCESS'
export const HOME_EDIT_WORKFLOW_TAGS_FAILURE = 'HOME_EDIT_WORKFLOW_TAGS_FAILURE'
export const HOME_EDIT_APP_TAGS_START = 'HOME_EDIT_APP_TAGS_START'
export const HOME_EDIT_APP_TAGS_SUCCESS = 'HOME_EDIT_APP_TAGS_SUCCESS'
export const HOME_EDIT_APP_TAGS_FAILURE = 'HOME_EDIT_APP_TAGS_FAILURE'
export const HOME_APPS_ATTACH_TO_START = 'HOME_APPS_ATTACH_TO_START'
export const HOME_APPS_ATTACH_TO_SUCCESS = 'HOME_APPS_ATTACH_TO_SUCCESS'
export const HOME_APPS_ATTACH_TO_FAILURE = 'HOME_APPS_ATTACH_TO_FAILURE'
export const HOME_APPS_COMPARISON_ACTION_START = 'HOME_APPS_COMPARISON_ACTION_START'
export const HOME_APPS_COMPARISON_ACTION_SUCCESS = 'HOME_APPS_COMPARISON_ACTION_SUCCESS'
export const HOME_APPS_COMPARISON_ACTION_FAILURE = 'HOME_APPS_COMPARISON_ACTION_FAILURE'
export const HOME_DELETE_APP_START = 'HOME_DELETE_APP_START'
export const HOME_DELETE_APP_SUCCESS = 'HOME_DELETE_APP_SUCCESS'
export const HOME_DELETE_APP_FAILURE = 'HOME_DELETE_APP_FAILURE'
export const HOME_APPS_MAKE_FEATURED_SUCCESS = 'HOME_APPS_MAKE_FEATURED_SUCCESS'
export const HOME_APPS_FETCH_APP_EXECUTIONS_START = 'HOME_APPS_FETCH_APP_EXECUTIONS_START'
export const HOME_APPS_FETCH_APP_EXECUTIONS_SUCCESS = 'HOME_APPS_FETCH_APP_EXECUTIONS_SUCCESS'
export const HOME_APPS_FETCH_APP_EXECUTIONS_FAILURE = 'HOME_APPS_FETCH_APP_EXECUTIONS_FAILURE'

export const HOME_APPS_RESET_MODALS = 'HOME_APPS_RESET_MODALS'
export const HOME_APPS_SHOW_MODAL = 'HOME_APPS_SHOW_MODAL'
export const HOME_APPS_HIDE_MODAL = 'HOME_APPS_HIDE_MODAL'
export const HOME_APPS_SET_FILTER_VALUE = 'HOME_APPS_SET_FILTER_VALUE'
export const HOME_APPS_RESET_FILTERS = 'HOME_APPS_RESET_FILTERS'

export const HOME_APPS_FEATURED_FETCH_START = 'HOME_APPS_FEATURED_FETCH_START'
export const HOME_APPS_FEATURED_FETCH_SUCCESS = 'HOME_APPS_FEATURED_FETCH_SUCCESS'
export const HOME_APPS_FEATURED_FETCH_FAILURE = 'HOME_APPS_FEATURED_FETCH_FAILURE'
export const HOME_APPS_FEATURED_TOGGLE_ALL_CHECKBOXES = 'HOME_APPS_FEATURED_TOGGLE_ALL_CHECKBOXES'
export const HOME_APPS_FEATURED_TOGGLE_CHECKBOX = 'HOME_APPS_FEATURED_TOGGLE_CHECKBOX'

// files page
export const HOME_FILES_FETCH_START = 'HOME_FILES_FETCH_START'
export const HOME_FILES_FETCH_SUCCESS = 'HOME_FILES_FETCH_SUCCESS'
export const HOME_FILES_FETCH_FAILURE = 'HOME_FILES_FETCH_FAILURE'
export const HOME_FILES_TOGGLE_ALL_CHECKBOXES = 'HOME_FILES_TOGGLE_ALL_CHECKBOXES'
export const HOME_FILES_TOGGLE_CHECKBOX = 'HOME_FILES_TOGGLE_CHECKBOX'
export const HOME_FILES_FETCH_FILE_DETAILS_START = 'HOME_FILES_FETCH_FILE_DETAILS_START'
export const HOME_FILES_FETCH_FILE_DETAILS_SUCCESS = 'HOME_FILES_FETCH_FILE_DETAILS_SUCCESS'
export const HOME_FILES_FETCH_FILE_DETAILS_FAILURE = 'HOME_FILES_FETCH_FILE_DETAILS_FAILURE'
export const HOME_FILES_DELETE_START = 'HOME_FILES_DELETE_START'
export const HOME_FILES_DELETE_SUCCESS = 'HOME_FILES_DELETE_SUCCESS'
export const HOME_FILES_DELETE_FAILURE = 'HOME_FILES_DELETE_FAILURE'
export const HOME_MAKE_PUBLIC_FOLDER_START = 'HOME_MAKE_PUBLIC_FOLDER_START'
export const HOME_MAKE_PUBLIC_FOLDER_SUCCESS = 'HOME_MAKE_PUBLIC_FOLDER_SUCCESS'
export const HOME_MAKE_PUBLIC_FOLDER_FAILURE = 'HOME_MAKE_PUBLIC_FOLDER_FAILURE'
export const HOME_RENAME_FILE_START = 'HOME_RENAME_FILE_START'
export const HOME_RENAME_FILE_SUCCESS = 'HOME_RENAME_FILE_SUCCESS'
export const HOME_RENAME_FILE_FAILURE = 'HOME_RENAME_FILE_FAILURE'
export const HOME_COPY_FILE_TO_SPACE_START = 'HOME_COPY_FILE_TO_SPACE_START'
export const HOME_COPY_FILE_TO_SPACE_SUCCESS = 'HOME_COPY_FILE_TO_SPACE_SUCCESS'
export const HOME_COPY_FILE_TO_SPACE_FAILURE = 'HOME_COPY_FILE_TO_SPACE_FAILURE'
export const HOME_FILES_MAKE_FEATURED_SUCCESS = 'HOME_FILES_MAKE_FEATURED_SUCCESS'
export const HOME_FILES_FETCH_SUBFOLDERS_SUCCESS = 'HOME_FILES_FETCH_SUBFOLDERS_SUCCESS'
export const HOME_LICENSE_ACTION_START = 'HOME_LICENSE_ACTION_START'
export const HOME_LICENSE_ACTION_SUCCESS = 'HOME_LICENSE_ACTION_SUCCESS'
export const HOME_LICENSE_ACTION_FAILURE = 'HOME_LICENSE_ACTION_FAILURE'

export const HOME_ADD_FOLDER_START = 'HOME_ADD_FOLDER_START'
export const HOME_ADD_FOLDER_SUCCESS = 'HOME_ADD_FOLDER_SUCCESS'
export const HOME_ADD_FOLDER_FAILURE = 'HOME_ADD_FOLDER_FAILURE'

export const HOME_FILES_ATTACH_TO_START = 'HOME_FILES_ATTACH_TO_START'
export const HOME_FILES_ATTACH_TO_SUCCESS = 'HOME_FILES_ATTACH_TO_SUCCESS'
export const HOME_FILES_ATTACH_TO_FAILURE = 'HOME_FILES_ATTACH_TO_FAILURE'

export const HOME_MOVE_FILE_START = 'HOME_MOVE_FILE_START'
export const HOME_MOVE_FILE_SUCCESS = 'HOME_MOVE_FILE_SUCCESS'
export const HOME_MOVE_FILE_FAILURE = 'HOME_MOVE_FILE_FAILURE'

export const HOME_FILES_RESET_MODALS = 'HOME_FILES_RESET_MODALS'
export const HOME_FILES_SHOW_MODAL = 'HOME_FILES_SHOW_MODAL'
export const HOME_FILES_HIDE_MODAL = 'HOME_FILES_HIDE_MODAL'
export const HOME_FILES_SET_FILTER_VALUE = 'HOME_FILES_SET_FILTER_VALUE'
export const HOME_FILES_RESET_FILTERS = 'HOME_FILES_RESET_FILTERS'

export const HOME_FILES_FEATURED_FETCH_START = 'HOME_FILES_FEATURED_FETCH_START'
export const HOME_FILES_FEATURED_FETCH_SUCCESS = 'HOME_FILES_FEATURED_FETCH_SUCCESS'
export const HOME_FILES_FEATURED_FETCH_FAILURE = 'HOME_FILES_FEATURED_FETCH_FAILURE'
export const HOME_FILES_FEATURED_TOGGLE_ALL_CHECKBOXES = 'HOME_FILES_FEATURED_TOGGLE_ALL_CHECKBOXES'
export const HOME_FILES_FEATURED_TOGGLE_CHECKBOX = 'HOME_FILES_FEATURED_TOGGLE_CHECKBOX'

export const HOME_SET_CURRENT_TAB = 'HOME_SET_CURRENT_TAB'
export const HOME_SET_CURRENT_PAGE = 'HOME_SET_CURRENT_PAGE'
export const HOME_FETCH_ACCESSIBLE_SPACES_SUCCESS = 'HOME_FETCH_ACCESSIBLE_SPACES_SUCCESS'
export const HOME_FETCH_ACCESSIBLE_SPACES_START = 'HOME_FETCH_ACCESSIBLE_SPACES_START'
export const HOME_FETCH_ACCESSIBLE_SPACES_FAILURE = 'HOME_FETCH_ACCESSIBLE_SPACES_FAILURE'
export const HOME_SELECT_ACCESSIBLE_SPACE = 'HOME_SELECT_ACCESSIBLE_SPACE'
export const HOME_SET_PAGE_COUNTERS = 'HOME_SET_PAGE_COUNTERS'
export const HOME_SET_INITIAL_PAGE_ADMIN_STATUS = 'HOME_SET_INITIAL_PAGE_ADMIN_STATUS'
export const HOME_FETCH_ATTACHING_ITEMS_SUCCESS = 'HOME_FETCH_ATTACHING_ITEMS_SUCCESS'
export const HOME_FETCH_ATTACHING_ITEMS_START = 'HOME_FETCH_ATTACHING_ITEMS_START'
export const HOME_FETCH_ATTACHING_ITEMS_FAILURE = 'HOME_FETCH_ATTACHING_ITEMS_FAILURE'
export const HOME_SET_IS_LEFT_MENU_OPEN = 'HOME_SET_IS_LEFT_MENU_OPEN'
export const HOME_DELETE_FILE_START = 'HOME_DELETE_FILE_START'
export const HOME_DELETE_FILE_SUCCESS = 'HOME_DELETE_FILE_SUCCESS'
export const HOME_DELETE_FILE_FAILURE = 'HOME_DELETE_FILE_FAILURE'
export const HOME_SELECT_ACCESSIBLE_LICENSE = 'HOME_SELECT_ACCESSIBLE_LICENSE'
export const HOME_FETCH_ACCESSIBLE_LICENSE_SUCCESS = 'HOME_FETCH_ACCESSIBLE_LICENSE_SUCCESS'
export const HOME_FETCH_ACCESSIBLE_LICENSE_START = 'HOME_FETCH_ACCESSIBLE_LICENSE_START'
export const HOME_FETCH_ACCESSIBLE_LICENSE_FAILURE = 'HOME_FETCH_ACCESSIBLE_LICENSE_FAILURE'
export const HOME_FETCH_COUNTERS_SUCCESS = 'HOME_FETCH_COUNTERS_SUCCESS'

export const HOME_ATTACH_LICENSE_START = 'HOME_ATTACH_LICENSE_START'
export const HOME_ATTACH_LICENSE_SUCCESS = 'HOME_ATTACH_LICENSE_SUCCESS'
export const HOME_ATTACH_LICENSE_FAILURE = 'HOME_ATTACH_LICENSE_FAILURE'

export const HOME_FETCH_FILES_BY_ACTION_START = 'HOME_FETCH_FILES_BY_ACTION_START'
export const HOME_FETCH_FILES_BY_ACTION_SUCCESS = 'HOME_FETCH_FILES_BY_ACTION_SUCCESS'
export const HOME_FETCH_FILES_BY_ACTION_FAILURE = 'HOME_FETCH_FILES_BY_ACTION_FAILURE'

export const HOME_EDIT_FILE_TAGS_START = 'HOME_EDIT_FILE_TAGS_START'
export const HOME_EDIT_FILE_TAGS_SUCCESS = 'HOME_EDIT_FILE_TAGS_SUCCESS'
export const HOME_EDIT_FILE_TAGS_FAILURE = 'HOME_EDIT_FILE_TAGS_FAILURE'
