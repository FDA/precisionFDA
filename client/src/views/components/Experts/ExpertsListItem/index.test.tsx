import React from 'react'
import { shallow } from 'enzyme'

import { ExpertsListItemBlogEntry, ExpertsListItemBlogEntrySmall, ExpertsListItemQuestionsAndAnswers } from '.'
import { IExpert } from '../../../../types/expert'

describe('ExpertsListItem test', () => {
  const getMockExpert = () => {
    const i = 1
    const mockExpert: IExpert = {
      id: i,
      user_id: i,
      title: 'Experts '+i,
      blogTitle: 'Blog Title '+i,
      blogPreview: 'Blog Preview '+i,
      blog: 'Blog '+i,
      about: '',
      image: '',
      state: 'open',
      scope: 'public',
      createdAt: new Date('March 1, 2021 20:21:00'),
      updatedAt: new Date('March 1, 2021 20:21:00'),
      totalAnswerCount: 10,
      totalCommentCount: 11,
    }
    return mockExpert
  }

  it('should render ExpertsListItemBlogEntry', () => {
    const mockExpert = getMockExpert()
    const wrapper = shallow(<ExpertsListItemBlogEntry expert={mockExpert} userCanEdit={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render ExpertsListItemBlogEntrySmall', () => {
    const mockExpert = getMockExpert()
    const wrapper = shallow(<ExpertsListItemBlogEntrySmall expert={mockExpert} userCanEdit={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render ExpertsListItemQuestionsAndAnswers', () => {
    const mockExpert = getMockExpert()
    const wrapper = shallow(<ExpertsListItemQuestionsAndAnswers expert={mockExpert} userCanEdit={false} />)

    expect(wrapper).toMatchSnapshot()
  })

})
