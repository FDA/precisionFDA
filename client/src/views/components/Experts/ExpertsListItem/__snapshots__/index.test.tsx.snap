// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertsListItem test should render ExpertsListItemBlogEntry 1`] = `
<div
  className="experts-list-item"
>
  <div
    className="experts-list-item__left-column"
  >
    <img
      alt="Profile icon for Experts 1's blog"
      className="expert-image img-circle"
      src=""
    />
  </div>
  <div
    className="experts-list-item__right-column experts-item-content"
  >
    <h1>
      Blog Title 1
    </h1>
    <div
      style={
        Object {
          "marginTop": "4px",
        }
      }
    >
      <span
        className="expert-name"
      >
        Experts 1
      </span>
      <span
        className="expert-date"
      >
        Mar 01, 2021
      </span>
    </div>
    <p>
      Blog Preview 1
    </p>
    <div
      className="experts-list-item__buttons pull-left"
      style={
        Object {
          "marginTop": "6px",
        }
      }
    >
      <Button
        className=""
        onClick={[Function]}
        size="md"
        type="default"
      >
        Expert Q&A
      </Button>
      <Link
        to="/experts/1"
      >
        <Icon
          fw={true}
          icon="fa-star-o"
        />
        <span
          aria-label="Click to view more information about Experts 1"
        >
          About This Expert
        </span>
      </Link>
      <Link
        to="/experts/1/blog"
      >
        <span
          aria-label="Click to read Experts 1's blog post"
        >
          Read Expert Blog Post ↗
        </span>
      </Link>
    </div>
  </div>
</div>
`;

exports[`ExpertsListItem test should render ExpertsListItemBlogEntrySmall 1`] = `
<div
  className="experts-list-item-small"
>
  <div
    className="experts-list-item-small__left-column"
  >
    <img
      alt="Profile icon for Experts 1's blog"
      className="expert-image img-circle"
      src=""
    />
  </div>
  <div
    className="experts-list-item-small__right-column experts-item-content"
  >
    <div
      className="expert-name"
    >
      Experts 1
    </div>
    <div
      className="expert-blog-title"
    >
      Blog Title 1
    </div>
    <div
      className="expert-date"
    >
      Mar 01, 2021
    </div>
  </div>
</div>
`;

exports[`ExpertsListItem test should render ExpertsListItemQuestionsAndAnswers 1`] = `
<div
  className="experts-list-item-small"
>
  <div
    className="experts-list-item-small__left-column"
  >
    <img
      alt="Profile icon for Experts 1's blog"
      className="expert-image img-circle"
      src=""
    />
  </div>
  <div
    className="experts-list-item-small__right-column experts-item-content"
  >
    <div
      className="expert-name"
    >
      <Link
        to="/experts/1"
      >
        <span>
          Experts 1
        </span>
      </Link>
    </div>
    <div
      className="answers-comments-container"
    >
      <div>
        <span>
          10
        </span>
         
        <span
          className="answers-comments"
        >
          Answers
        </span>
      </div>
      <div>
        <span>
          11
        </span>
         
        <span
          className="answers-comments"
        >
          Comments
        </span>
      </div>
    </div>
  </div>
</div>
`;
